Add sdl2 samples

bucanero opened the pull request:
Ref https://github.com/pspdev/pspsdk/pull/119

sharkwouter commented on sdl2/PSPBUILD:
Please increase the pkgrel when making changes to a PSPBUILD

sharkwouter commented on sdl2-mixer/PSPBUILD:
```diff
@@ -30,9 +40,13 @@ build() {
 
 package() {
+    mkdir -p "$pkgdir/psp/sdk/samples/sdl2/mixer/"
+    mv main.c "$pkgdir/psp/sdk/samples/sdl2/mixer/"
```
I would use the `install` command instead of `mv`. Make sure to get the files from `$srcdir/`. Right now these files are 0 byte links instead of the actual files.

sharkwouter reviewed (changes requested):
Thanks for the PR, overall it looks pretty good. I see the following problems which should be addressed before it could be merged:
- The pkgrel in the PSPBUILD files needs to be increased, otherwise psp-pacman will not retrieve the updated versions.
- The `mv` command should be replaced with `install`. Example `install -m 644 $srcdir/main.c $pkgdir/psp/sdk/samples/sdl2/basic/main.c`.
- Use `$srcdir/main.c` when copying files instead of `main.c`. The packages now contain empty files.
- Move the sample files to their own directory within the directory the PSPBUILD is in. Now it seems like the packages have custom CMakeLists.txt and main.c files for building sdl2 itself.
- Included the license for the font and sound files.

Some things which I will leave up to you if you want to change them or not:
- These libraries are not technically part of the pspsdk, so I'm not sure if the samples are being put in the right place, although this is probably where people will look for samples. I'm 50/50 on keeping it like this or changing it. You decide.

Again, thanks for making this PR. You're doing great work here. Looking forward to merging this.

bucanero commented:
hi @sharkwouter , thanks for the feedback

I have updated the PR and moved the example files to a `sample` folder under each lib, but it fails when downloading the sources. This is how I defined the sources:

```
source=(
    "git+https://github.com/libsdl-org/SDL.git#commit=428b5ae5464d567881c383edeb2d1e5712b18ffd"
    "sample/main.c"
    "sample/CMakeLists.txt"
)
```

then the error:
```
==> ERROR: main.c was not found in the build directory and is not a URL.
```

do you know the right way to define the source files under the `sample` folder?

About the licenses, I'll check as I was just using the samples from the pspdev website guide. If I can't find the licenses I'll just replace the ogg/ttf with other public non-restricted files.

sharkwouter commented:
That's a shame, maybe instead it would make sense to name them something which makes it more obvious that they are a sample instead, then rename them when installing.

bucanero commented:
> That's a shame, maybe instead it would make sense to name them something which makes it more obvious that they are a sample instead, then rename them when installing.

I've renamed the files with `.sample` extension, and renamed them to the original filename when installing.

The only issue right now is that the `eigen3` package is failing to build, but this PR has nothing to do with that package. For some reason the sha256 of that package is failing the verification. Maybe the source .tgz was updated? 🤔

sharkwouter commented:
I'll try to fix the eigen3 failure. Seems Gitlab release archives cannot be relied on for packaging: https://forum.gitlab.com/t/gitlab-release-tarball-stability/41888/2

bucanero commented:
> I just restarted the test and it works, surprisingly. Not sure what happened. Thanks for the work on this! I've merged it now!

oh that was weird! well happy to hear everything worked in the end! 😄
